ships/0038-release-branching-backports.md

```diff
@@ -0,0 +1,200 @@
+- Apply the backport process to "alpha" maturity projects in Shipwright (ex:
+  triggers).
+- Docs/website content strategy for released versions.
+- Apply backport process to v0.11.0 and earlier.
+- Prescribe a cadence for feature and bugfix releases.
+
+## Proposal
+
+Shipwright projects will continue to use the `main` branch for feature
+development. When the project maintainers decide to create a release, they **MUST**
+create a branch that adheres to the `release-vX.Y` naming convention. `X.Y`
+represents the semantic _major_ and _minor_ version of the given release,
+respectively.
+
+Release tags and artifacts for a given `X.Y` version **MUST** be created from the
+respective `release-vX.Y` release branch. Releases **MUST** use a valid semantic
+version when tagging or versioning artifacts, of the format `vX.Y.Z`. Projets
+**MAY** append a hyphen to indicate a release candidate or other form of "pre-
+release" version, in accorance with the SemVer 2.0 specification. For example,
+`v0.13.0-rc2` can be used to represent version 0.13.0, _release candidate 2_.
+CI checks to merge code in the the `main` branch **MUST** also be run against code
+in release branches. Other measures applied to `main` branches (ex: branch
+protection rules) **MUST** also be applied to release branches.
+
+All code changes **SHOULD** be merged into the `main` branch before being applied
+to a previous release branch, or a release branch that is issuing release
+candidates. Merging directly into a release branch without an equivalent code
+change in the `main` branch is allowed at sole the discretion of project
+maintainers. Once code is merged in `main`, it is eligible to be backported to an
+appropriate release branch using `git cherry-pick` or equivalent commands.
+
+Backport code changes **SHOULD** be merged into release branches in descending
+semantic version order if the code change should be applied to more than one release
+version. This ensures upgrades from one patch version to the next minor version do
+not introduce regressions with respect to bugs or security fixes. For example, if
+the `release-v0.13` branch is issuing release candidates, and a bug fix for the
+release needs to be backported to `release-v0.12`, code should merge in the
+following order:
+
+1. Merge fix to `main`
+2. Cherry pick to `release-v0.13`
+3. Cherry pick to `release-v0.12` _after_ code merges in `release-v0.13`.
+
+Merging out of semantic version order **MAY** be done at the discretion of project
+maintainers, especially in situations that involve security advisories with
+"Critical" severity ratings.
+
+### User Stories [optional]
+
+- As a maintainer of a Shipwright distribution, I want Important/Critical severity
+  security advisories to be backported to a `v0.12.z` release stream so that I can
+  patch security vulnerabilities without introducing new features or breaking
+  changes.
+- As a cluster admin / architect deploying Shipwright to my engineering teams, I
```

Comment:
Can you generalize this. We should for the time being only provide such fixes for the latest released version. Eventually having something like LTS releases where we maintain certain releases for a longer time would be a separate ship.

Reply:
Done! I also tried to clarify that LTS is out of scope for this proposal.